### DISCUSSION TITLE
XEP-0425: on refresh, see messages that the moderator had retracted

### DISCUSSION THREAD

**Discussion by Quit4944** [processone/ejabberd#4034] [Questions] [answered]
Hi! I have a test environment with ejabberd 23.04 and converse 10.0.0, I wanted to try the long awaited XEP-0425 and ... it works!
But something happens and I would like to understand if it's normal or if I have to fix it in some way.

In converse.js, i'm using the `clear-messages-on-reconnection` option ([https://conversejs.org/docs/html/configuration.html?#clear-messages-on-reconnection](url)) and this is what happens:
- If true, when a moderator retracts a message, the message is retracted correctly, but when refresh the page (e.g. exit and re-enter the chat or press F5) I see in the chat the retracted message again in plaintext, as if it had not been retracted/deleted;

- If false, it works fine: the moderator retracts the message, and when do a refresh, the message remain retracted/deleted in the chat.

Why? what can i do to fix? is it a bug or did I make some setting wrong? (I need to keep clear-messages-on-reconnection=true, because I delete the messages with "delete_old_mam_messages" command to clean the chat).

At the refresh, it seems to go to read and retrieve the messages from the database, and it seems that these messages have not been retracted in the database, but are still present, and therefore converse shows them in plaintext. It's correct?

I tried to modify many options, both those of mod_mam by ejabberd and all those of converse, many hours of tests, but the result is always the same: the key-option of the whole game is clear-messages-on-reconnection, in my case, and I don't know how to fix.

please any suggestions, I want to keep clear-messages-on-reconnection=true and I want messages to remain retracted and not visible on refresh.

My environment test:
Windows 10 with Ejabberd on Docker Desktop

Ejabberd:
Mnesia (auth_method = internal), websocket, a MUC with mod_mam active

Converse:
authentication: "login"
view_mode: "fullscreen"
auto_login: true,
keepalive: false,
auto_join_rooms: on my MUC
archived_messages_page_size: 50
muc_history_max_stanzas: 0
message_archiving: undefined
clear_messages_on_reconnection: true/false ! <- this makes the difference

**Comment by prefiks** [comment]
This should remove messages both from in memory room history and from those stored by mod_mam in db, i am pretty sure that was the case when i tested it. Could you attach your ejabberd config or at least configuration of mod_mam that you use and default_db_type if you have this configured?  Possibly there is some issue with configuration like yours (and i would like to test that), or could it be that conversejs have its own copy of messages that don't get purged on retraction.

**Comment by Quit4944** [comment]
Ok, here the "almost original" ejabberd and converse config:
[converse_cfg.js.txt](https://github.com/processone/ejabberd/files/11517400/converse_cfg.js.txt)
[ejabberd.yml.txt](https://github.com/processone/ejabberd/files/11517401/ejabberd.yml.txt)

Note: 
- i'm not using the _mod_conversejs_ module, but i create the chat with _converse.initialize ({....})_

- when migrated from 22.10 to 23.04, I didn't run any script to update the tables of my db mnesia, because I read that it updated automatically.

**Reply by Quit4944** [reply, marked as answer]
hey @prefiks, I found something useful: the problem is Firefox! on Chrome it works fine.
Do you think Firefox will be supported for this fix? I have tried both the Firefox Stable version and the Developer Edition, and both have the problem; And then: is in Ejabberd or in Converse (or in Firefox) the bug? Do I need to report this to Converse?

**Comment by prefiks** [comment]
So if it works in one browser but not other, then this looks like bug in client or browser - ejabberd doesn't sends anything different to different browser, we don't even have information about that in server. I would suggest reporting this in converse.
